src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs

```diff
@@ -305,22 +311,37 @@ private async Task ListenForMessages(CancellationToken cancellationToken)
 
             if (handlerToAwait != null)
             {
-                try
+                if (this.ParallelMessageProcessing)
                 {
-                    await handlerToAwait;
+                    var task = Task.Run(async () =>
```

Comment:
Just call RunTask without awaiting, no reason to wrap this if we're not doing anything with it.

Reply:
there will be a build warning, to get rid of it, you need to either await it or assign it to a variable like the `var task = ` here.

![image](https://user-images.githubusercontent.com/13777222/166614779-172cd0af-a755-40e5-a51b-4975758367a1.png)

Reply:
You should also be able to add a suppression for it (VS is much better for this - VS Code doesn't have the auto-fix options available)

In the end it's just a warning though so as long as we make it clear that we intend now to await it I don't see a problem with that.

(you could also at the very least just set a var for the original task `var task = RunTask(...);`)

Reply:
Oh yeah, that's a thing too 😁